modules/core/common/createReduxStore.ts

```diff
@@ -8,11 +8,63 @@ export const getStoreReducer = (reducers: any) =>
     ...reducers
   });
 
-const createReduxStore = (reducers: Reducer, initialState: DeepPartial<any>, routerMiddleware?: Middleware): Store => {
+const requestMiddleware: Middleware = _state => next => action => {
+  const { types, APICall, ...rest } = action;
+  if (!types) {
+    return next(action);
+  }
+
+  const { REQUEST, SUCCESS, FAIL } = types;
+
+  next({ type: REQUEST || null, ...rest });
+
+  const handleAPICall = async () => {
+    try {
+      const result = await APICall();
+      const data = result && result.data;
+      if (data.errors) {
+        throw { response: result };
+      }
+      next({
+        ...rest,
+        type: SUCCESS || null,
+        payload: data
+      });
+
+      return data;
+    } catch (e) {
+      if (e.response && e.response.data && e.response.data.status === 401) {
+        return next({ ...action, type: null, status: e.response.data.status });
+      }
+      const data = e.response && e.response.data;
+      next({
+        ...rest,
+        type: FAIL || null,
+        payload: data
+      });
+      throw e;
+    }
+  };
+
+  return handleAPICall();
+};
+
+const createReduxStore = (
+  reducers: Reducer,
+  initialState: DeepPartial<any>,
+  routerMiddleware?: Middleware,
+  reduxMiddlewares?: Middleware[]
```

Comment:
Isn't it easier and cleaner to use default values?

Reply:
Fixed